src/components/Tokens/index.tsx

```diff
@@ -0,0 +1,151 @@
+
+import * as React from 'react';
+import {
+  useSelector
+} from 'react-redux';
+
+import TokenSelector from './TokenSelector';
+import { SELECT_VARIANTS } from 'components/Select';
+
+import {
+  WrappedToken,
+  WRAPPED_TOKEN,
+  WRAPPED_TOKEN_SYMBOL,
+  WrappedTokenLogoIcon,
+  CollateralToken,
+  COLLATERAL_TOKEN,
+  COLLATERAL_TOKEN_SYMBOL,
+  CollateralTokenLogoIcon,
+  GovernanceToken,
+  GOVERNANCE_TOKEN,
+  GOVERNANCE_TOKEN_SYMBOL,
+  GovernanceTokenLogoIcon
+} from 'config/relay-chains';
+import { displayMonetaryAmount } from 'common/utils/utils';
+import {
+  TokenType,
+  StoreType
+} from 'common/types/util.types';
+
+interface TokenOption {
+  token: WrappedToken | CollateralToken | GovernanceToken;
+  type: TokenType;
+  balance: string;
+  symbol: string;
+  icon: JSX.Element;
+}
+
+interface Props {
+  variant?: typeof SELECT_VARIANTS[number];
+  callbackFunction?: (token: TokenOption) => void;
+  showBalances?: boolean;
+}
+
+const Tokens = ({
+  variant = 'optionSelector',
+  callbackFunction,
+  showBalances = true
+}: Props): JSX.Element => {
+  const [tokenOptions, setTokenOptions] = React.useState<Array<TokenOption> | undefined>(undefined);
+  const [currentToken, setCurrentToken] = React.useState<TokenOption | undefined>(undefined);
```

Comment:
@tomjeatt 
One question out of curiosity. Do we need `undefined` explicitly? I think it's undefined by default.

Reply:
Good question :) I like setting undefined because it removes any ambiguity over whether it could be null. I also think if you're using a union type (`TokenOption | undefined` in this case) then it's a bit clearer to set undefined. Finally, it also makes it little bit clearer for anyone coming into the project if they're not aware that undefined is set by default (i.e. not much React/TS experience).